### ISSUE TITLE
Error (ein): Undo failure diagnostic prevents me from using undo

### ISSUE THREAD

**Issue by bvborisov** [millejoh/emacs-ipython-notebook#649]
## Problem description
Quite often when I use ein I get some sort of exception from emacs and I can no longer do undo. Full stack trace at the end of the bug report

## Steps to reproduce the problem
Nothing in particular, just editing my notebook.

## System info:

```cl
("EIN system info"
 :emacs-version
 "GNU Emacs 25.2.2 (x86_64-pc-linux-gnu, GTK+ Version 3.22.21)
 of 2017-09-22, modified by Debian"
 :emacs-bzr-version nil
 :window-system x
 :emacs-variant nil
 :os
 (:uname
  "Linux x-Inspiron-5558 4.15.0-72-generic #81-Ubuntu SMP Tue Nov 26 12:20:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
"
  :lsb-release
  "No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic
")
 :notebook
 "Name: notebook
Version: 6.0.1
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /home/x/anaconda3/envs/env_full/lib/python3.7/site-packages
Requires: traitlets, prometheus-client, ipykernel, ipython-genutils, jupyter-client, Send2Trash, jinja2, jupyter-core, nbformat, tornado, pyzmq, nbconvert, terminado
Required-by: widgetsnbextension, jupyterlab, jupyterlab-server, jupyter, jupyter-nbextensions-configurator, jupyter-latex-envs, jupyter-contrib-nbextensions, jupyter-contrib-core
"
 :ipython "7.8.0
"
 :image-types
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :image-types-available
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :request
 (:backend curl)
 :ein
 (:version "20200127.249"
	   :source-dir "/home/x/.emacs.d/elpa/ein-20200127.249/")
 :lib
 ((:name "websocket"
	 :path "~/.emacs.d/elpa/websocket-20191123.2208/websocket.elc"
	 :featurep t
	 :version-var websocket-version
	 :version "1.12")
  (:name "auto-complete"
	 :path "~/.emacs.d/elpa/auto-complete-20170125.245/auto-complete.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "request"
	 :path "~/.emacs.d/elpa/request-20190923.1502/request.elc"
	 :featurep t
	 :version-var request-version
	 :version "0.3.0")
  (:name "deferred"
	 :path "~/.emacs.d/elpa/deferred-20170901.1330/deferred.elc"
	 :featurep t
	 :version-var deferred:version
	 :version "0.5.0")
  (:name "polymode"
	 :path "~/.emacs.d/elpa/polymode-20190714.2017/polymode.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "markdown-mode"
	 :path "~/.emacs.d/elpa/markdown-mode-20190802.2215/markdown-mode.elc"
	 :featurep t
	 :version-var markdown-mode-version
	 :version "2.4-dev")
  (:name "dash"
	 :path "~/.emacs.d/elpa/dash-20190920.1035/dash.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "s"
	 :path "~/.emacs.d/elpa/s-20180406.808/s.elc"
	 :featurep t
	 :version-var nil
	 :version nil)
  (:name "skewer-mode"
	 :path "~/.emacs.d/elpa/skewer-mode-20180706.1807/skewer-mode.elc"
	 :featurep t
	 :version-var nil
	 :version nil)))
```

## Full stack trace from ein itself:
```
Error (ein): Undo failure diagnostic (nil (23239 . 23243) nil (a . -23239) (#<marker at 23241 in *ein: http://127.0.0.1:8890/Machine Learning Problem Sheet 1.ipynb*> . -1) (s . -23240) (#<marker at 23241 in *ein: http://127.0.0.1:8890/Machine Learning Problem Sheet 1.ipynb*> . -1) (d . -23241) (#<marker at 23241 in *ein: http://127.0.0.1:8890/Machine Learning Problem Sheet 1.ipynb*> . -1) 23242 nil (23241 . 23242) (t . 0) (23239 . 23241) (t . 0) nil (23235 . 23239) (t . 0) nil (p . -23235) (r . -23236) nil (23235 . 23237) nil (23234 . 23235) nil (23221 . 23234) nil (23208 . 23221) nil (23187 . 23208) nil (23186 . 23187) nil (23167 . 23169) nil (3 . -23167) nil (23192 . 23193) nil (23254 . 23261) nil (predictions3 . -23254) nil (23244 . 23250) nil (Henry' . -23244) nil (s  . -23250) nil (simple  . -23252) nil (predictions . -23259) nil (lr.fit(X_tr, y_tr)
predictions3 = lr.predict(X_te)
 . 23237) (t . 0) nil (23094 . 23095) nil (1 . -23094) (t . 0) nil (23086 . 23095) nil (23177 . 23186) nil (x . -23177) (  . -23178) nil (23177 . 23179) (t . 0) nil (23207 . 23209) nil (e . -23207) nil (23205 . 23208) nil (23198 . 23202) nil (23193 . 23195) (t . 0) nil (nil rear-nonsticky nil 23087 . 23088) (23062 . 23088) nil (23060 . 23062) (t . 0) nil (lm = linear_model.Lasso()
 . 23181) nil (clf . -23165) nil (23076 . 23078) nil (e . -23076) nil (22864 . 22865) nil (22831 . 22832) nil (22824 . 22825) nil (22463 . 22464) nil (22450 . 22451) nil (22437 . 22438) nil (22403 . 22404) nil (22375 . 22376) nil (22346 . 22347) nil (23067 . 23068) nil (23065 . 23067) (t . 0) nil (23003 . 23170) nil (Linear . -23003) nil (Regression . -23009) nil (nil rear-nonsticky nil 23018 . 23019) (23003 . 23019) nil (23003 . 23004) nil (23021 . 23026) nil (l . -23021) (a . -23022) (s . -23023) (s . -23024) nil (23021 . 23025) nil (Linear . -23021) nil (Regression . -23027) (t . 0) nil (23004 . 23005) nil (r . -23004) (t . 0) nil (23003 . 23142) nil (
 . -22908) (
 . -22909) (t . 0) nil (22908 . 22910) nil (22872 . 22879) (t . 0) nil (22788 . 22794) nil (Logistic . -22788) nil (22901 . 22903) nil () . -22901) nil (22901 . 22902) nil (pred . -22890) (22894 . 22905) nil (22888 . 22894) nil (22867 . 22888) nil (predicti . -22867) nil (22860 . 22875) nil (22859 . 22860) nil (22832 . 22843) nil (s . -22832) nil (22828 . 22833) nil (22836 . 22844) nil (22828 . 22836) nil (22827 . 22828) nil (22809 . 22827) nil (22808 . 22809) nil (22807 . 22808) nil (22791 . 22807) nil (22770 . 22791) nil (l . -22770) nil (22770 . 22771) nil (5556 . 5557) nil (817231 . -5556) nil (5561 . 5562) nil (x . -5561) nil (5561 . 5562) nil (i . -5561) (3 . -5562) nil (5560 . 5563) nil (1 . -5560) (t . 0) nil (5556 . 5561) nil (1 . -5556) (2 . -5557) (4 . -5558) nil (5558 . 5559) nil (3 . -5558) nil (5556 . 5559) nil (2 . -5556) nil (5556 . 5557) nil (1 . -5556) nil (5556 . 5557) nil (0 . -5556) nil (5539 . 5541) nil (0 . -5539) (1 . -5540) nil (5539 . 5541) nil (2 . -5539) (5 . -5540) (t . 0) nil (# . -8689) nil (8673 . 8674) nil (8669 . 8673) nil (
 . -8669) nil (     . -8670) nil (8588 . 8589) nil (8499 . 8500) nil (8474 . 8475) nil (8421 . 8422) nil (8617 . 8620) nil (8529 . 8532) nil (V . -8529) nil (8529 . 8530) (t . 0) nil (8595 . 8596) nil (_ . -8595) nil (best . -8596) nil (8510 . 8511) nil (_ . -8510) nil (best . -8511) nil (nil rear-nonsticky nil 8666 . 8667) (8493 . 8667) nil (8492 . 8493) nil (8417 . 8493) nil (# . 8421) nil (8437 . 8438) (t . 0) nil (# . -8437) nil (8421 . 8422) nil (    lm2 = linear_model.LassoCV(alphas=alphas, cv=5)
    lm2.fit(X_tr, y_tr)
 . 8417) nil (8506 . 8507) nil (_ . -8506) nil (best . -8507) nil (8516 . 8517) nil (nil rear-nonsticky nil 8529 . 8530) (8512 . 8530) nil (8511 . 8512) nil (nil rear-nonsticky nil 8510 . 8511) (8493 . 8511) nil (    return lm_best . 8493) (t . 0) nil (8488 . 8492) nil (8478 . 8488) nil (t . -8478) nil (8473 . 8479) nil (8421 . 8427) nil (8463 . 8467) nil (8462 . 8463) nil (8457 . 8462) nil (8442 . 8457) nil (n_ . -8442) nil (alphas . -8444) nil (8447 . 8450) nil (j . -8447) (a . -8448) (s . -8449) nil (8441 . 8450) nil (8423 . 8434) nil (8421 . 8423) nil (8421 . 8428) nil (8417 . 8421) nil (8416 . 8417) nil (5556 . 5557) nil (42 . -5556) nil (5539 . 5541) nil (7 . -5539) (5 . -5540) nil (5701 . 5702) nil (5693 . 5697) nil (5690 . 5691) nil (5685 . 5686) nil (* . -5685) nil (5682 . 5686) nil (5679 . 5680) nil (5671 . 5675) nil (5668 . 5669) nil (5660 . 5664) nil (5662 . 5683) nil (5660 . 5662) nil (x_ . -5660) nil (tr,  . -5662) nil (5654 . 5666) nil (5558 . 5559) nil (5548 . 5558) nil (5527 . 5548) nil (5521 . 5527) nil (5495 . 5505) nil (5488 . 5495) nil (a . -5488) nil (5486 . 5489) nil (X_ . -5486) nil (tr,  . -5488) nil (5491 . 5492) nil (5486 . 5491) nil (5486 . 5502) nil (5486 . 5487) nil (
 . -5419) nil (5418 . 5419) nil (22021 . 22022) nil (21994 . 21995) nil (21966 . 21967) nil (3 . -21963) nil (21664 . 21665) nil (21654 . 21655) nil (21961 . 21962) nil (22057 . 22059) nil (  . -22057) nil (22053 . 22058) nil (22043 . 22047) nil (b . -22043) (" . -22044) nil (22041 . 22045) nil (22035 . 22052) nil (x . -22035) nil (22029 . 22036) nil (. . -22017) nil (22015 . 22018) nil (22003 . 22015) nil (22002 . 22003) nil (21972 . 21973) nil (21976 . 21977) nil (X . -21976) nil (nil rear-nonsticky nil 22000 . 22001) (21976 . 22001) nil (21975 . 21976) nil (nil rear-nonsticky nil 21974 . 21975) (21950 . 21975) nil (X_tr = data_tr.iloc[:,-1] . 21950) nil (21969 . 21975) nil (21964 . 21969) nil ([ . -21964) nil (21950 . 21965) nil (21662 . 21663) nil (r . -21662) nil (21653 . 21654) nil (e . -21653) nil (21647 . 21666) nil (21628 . 21630) nil (21607 . 21628) nil (p . -21607) (r . -21608) nil (21607 . 21609)) (aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 07580 07580 07580 07580 07580 07580 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f de236 de236 de236 de236 de236 de236 de236 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 aab20 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 07580 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 daca7 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 1b5b1 42d45 42d45 42d45 de236 de236 de236 de236 de236 de236 de236 de236 de236 68192 68192 68192 68192 de236 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f 7ce3f de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 de236 68192 68192 68192 68192 68192 68192 68192 68192 68192 68192 3b07a 3b07a 3b07a 3b07a 3b07a 3b07a 3b07a 3b07a 3b07a) | 19

```

**Comment by dickmao** [comment]
I made some fixes in 5a23c06d 131a4d05 922c5b6b 5e9c30e1.  Perhaps you could try reinstalling the latest from MELPA.

Undo is difficult unfortunately, and I find myself having to constantly fix it.  Sorry.

**Comment by bvborisov** [comment]
Hey, no worries. I am already running `  ein                20200127.249  dependency            Emacs IPython Notebook` 
Is this the latest version available?

**Comment by dickmao** [comment]
20200202.218 is the latest.

**Comment by veprbl** [comment]
I've come up with steps to reproduce a similar issue. I've filed a separate #670 because it could be a different bug.

**Comment by veprbl** [comment]
This bug is not completely gone yet. Unfortunately, I have not yet able to come up with the new steps for reproduction.

**Comment by dickmao** [comment]
I believe it.  Alas, most undo failure modes are "random" and will depend on temporal circumstances.  It would be helpful just to report the full output of the "Undo failure diagnostic," as that would at least give me an intuition about what went wrong.

**Comment by veprbl** [comment]
I've got something:
```
Error (ein): Undo failure diagnostic (nil (apply activate-cursor-for-undo 20) (1198 . 1199) (apply deactivate-cursor-after-undo 20) (1092 . 1093) nil (apply activate-cursor-for-undo 20) (1143 . 1144) (apply deactivate-cursor-after-undo 20) nil (1038 . 1039) nil (969 . 970) nil (917 . 918) nil (cos( . -1112) (1116 . 1131) (cos( . -1060) (1064 . 1079) (cos( . -939) (943 . 958) nil (sin( . -1131) (1135 . 1150) nil (sin( . -997) (1001 . 1016) nil (sin( . -888) (892 . 907) nil (786 . 796) nil (points["phi"] . 786) nil (1172 . 1177) nil (1167 . 1168) nil (= . -1167) (= . -1168) nil (, . -1219) (  . -1220) (l . -1221) (e . -1222) (n . -1223) (g . -1224) (t . -1225) (h . -1226) (= . -1227) (0 . -1228) (. . -1229) (1 . -1230) nil (] . -1173) nil ([ . 1153) nil (1150 . 1153) nil (1163 . 1172) nil (1160 . 1163) nil (o . -1160) (h . -1161) (i . -1162) (" . -1163) nil (1158 . 1164) nil (p . -1158) nil (1152 . 1159) nil (i . -1152) (n . -1153) (t . -1154) nil (1149 . 1155) nil (1147 . 1150) nil (1191 . 1192) nil (1189 . 1191) nil (1183 . 1186) nil (1177 . 1180) nil (f . -1174) nil (1171 . 1175) nil (1165 . 1168) nil (1160 . 1162) nil (c . -1160) nil (1159 . 1161) nil () . -799) nil (set( . -786) nil (803 . 804) nil (786 . 790) nil (n . -786) (p . -787) nil (786 . 788) nil (791 . 799) nil (s . -791) (. . -792) (p . -793) (h . -794) (i . -795) ([ . -796) (" . -797) nil (796 . 798) nil ([ . -796) nil (786 . 797) nil (721 . 723) nil (710 . 714) nil (715 . 717) nil (d . -715) (n . -716) nil (706 . 717) nil (705 . 706) nil (680 . 681) nil (672 . 679) nil (nil rear-nonsticky nil 696 . 697) (nil fontified nil 655 . 697) (655 . 697) nil (# . -1067) nil (1020 . 1023) nil (r . -1020) nil (nil rear-nonsticky nil 1046 . 1047) (nil fontified nil 1045 . 1047) (nil fontified nil 1044 . 1045) (nil fontified nil 1041 . 1044) (nil fontified nil 1040 . 1041) (nil fontified nil 1022 . 1040) (nil fontified nil 1021 . 1022) (nil fontified nil 1019 . 1021) (nil fontified nil 1018 . 1019) (nil fontified nil 1011 . 1018) (1011 . 1047) nil (1008 . 1011) nil (938 . 941) nil (r . -938) nil (nil rear-nonsticky nil 964 . 965) (nil fontified nil 963 . 965) (nil fontified nil 962 . 963) (nil fontified nil 959 . 962) (nil fontified nil 958 . 959) (nil fontified nil 940 . 958) (nil fontified nil 939 . 940) (nil fontified nil 937 . 939) (nil fontified nil 936 . 937) (nil fontified nil 929 . 936) (929 . 965) nil (927 . 929) nil (  . -927) nil (927 . 928) nil (926 . 927) nil (980 . 982) nil (z . -980) nil (939 . 941) nil (r . -939) nil (898 . 900) nil (r . -898) nil (966 . 967) nil (z . -966) nil (926 . 927) nil (y . -926) nil (886 . 887) nil (x . -886) nil (nil fontified nil 980 . 981) (nil fontified nil 979 . 980) (nil fontified nil 978 . 979) (nil fontified nil 977 . 978) (nil fontified nil 970 . 977) (nil fontified nil 967 . 970) (nil fontified nil 966 . 967) (nil fontified nil 965 . 966) (nil fontified nil 963 . 965) (nil fontified nil 962 . 963) (nil fontified nil 959 . 962) (nil fontified nil 958 . 959) (nil fontified nil 940 . 958) (nil fontified nil 939 . 940) (nil fontified nil 938 . 939) (nil fontified nil 937 . 938) (nil fontified nil 927 . 937) (nil fontified nil 926 . 927) (nil fontified nil 925 . 926) (nil fontified nil 923 . 925) (nil fontified nil 922 . 923) (nil fontified nil 919 . 922) (nil fontified nil 918 . 919) (nil fontified nil 900 . 918) (nil fontified nil 899 . 900) (nil fontified nil 898 . 899) (nil fontified nil 897 . 898) (nil fontified nil 887 . 897) (nil fontified nil 886 . 887) (886 . 981) nil (884 . 886) nil (850 . 853) nil (sin . -850) nil (829 . 830) nil (x . -829) nil (nil rear-nonsticky nil 868 . 869) (nil fontified nil 868 . 869) (nil fontified nil 866 . 868) (nil fontified nil 865 . 866) (nil fontified nil 862 . 865) (nil fontified nil 861 . 862) (nil fontified nil 843 . 861) (nil fontified nil 842 . 843) (nil fontified nil 841 . 842) (nil fontified nil 840 . 841) (nil fontified nil 830 . 840) (nil fontified nil 829 . 830) (829 . 869) nil (nil rear-nonsticky nil 828 . 829) (nil fontified nil 828 . 829) (nil fontified nil 826 . 828) (nil fontified nil 825 . 826) (nil fontified nil 822 . 825) (nil fontified nil 821 . 822) (nil fontified nil 803 . 821) (nil fontified nil 802 . 803) (nil fontified nil 801 . 802) (nil fontified nil 800 . 801) (nil fontified nil 790 . 800) (nil fontified nil 789 . 790) (789 . 829) nil (
 . 789) nil (x = points["r"] * np.sin(points["phi"]) . 789) nil (807 . 814) nil (818 . 821) nil (] . -818) nil (801 . 819) nil (794 . 801) nil (i . -794) nil (789 . 795) nil (788 . 789) nil (nil rear-nonsticky nil 803 . 804) (nil fontified nil 803 . 804) (nil fontified nil 802 . 803) (nil fontified nil 801 . 802) (nil fontified nil 800 . 801) (nil fontified nil 793 . 800) (793 . 804) nil (789 . 793) nil (787 . 789) nil (562 . 563) nil (560 . 561) nil (559 . 562) nil (. . -559) (z . -560) nil (559 . 561) nil (536 . 537) nil (522 . 524) nil (phi . -522) nil (508 . 510) nil (phi . -508) nil (nil rear-nonsticky nil 548 . 549) (nil fontified nil 548 . 549) (nil fontified nil 547 . 548) (nil fontified nil 545 . 547) (nil fontified nil 544 . 545) (nil fontified nil 542 . 544) (nil fontified nil 541 . 542) (nil fontified nil 538 . 541) (nil fontified nil 537 . 538) (nil fontified nil 534 . 537) (534 . 549) nil (nil rear-nonsticky nil 533 . 534) (nil fontified nil 533 . 534) (nil fontified nil 532 . 533) (nil fontified nil 530 . 532) (nil fontified nil 529 . 530) (nil fontified nil 527 . 529) (nil fontified nil 526 . 527) (nil fontified nil 523 . 526) (nil fontified nil 522 . 523) (nil fontified nil 519 . 522) (519 . 534) nil (nil rear-nonsticky nil 518 . 519) (nil fontified nil 518 . 519) (nil fontified nil 517 . 518) (nil fontified nil 515 . 517) (nil fontified nil 514 . 515) (nil fontified nil 512 . 514) (nil fontified nil 511 . 512) (nil fontified nil 508 . 511) (nil fontified nil 507 . 508) (nil fontified nil 504 . 507) (504 . 519) nil (485 . 486) nil (i . -485) nil (500 . 502) nil (i . -500) (8 . -501) nil (493 . 496) nil (h . -493) nil (493 . 494) nil (z . -493) nil (467 . 468) nil (z . -467) nil (480 . 481) nil (467 . 468) nil (x . -467) nil (nil rear-nonsticky nil 502 . 503) (nil fontified nil 455 . 503) (455 . 503) nil (450 . 455) (431 . 449) (431 . 432) nil (430 . 431) nil (429 . 430) nil (432 . 438) nil (np.array(points) . 432) nil (429 . 430) nil (385 . 394) nil (422 . 438) nil (    x = r +  . 422) nil (
 . 422) nil (for r, z, phi, br, bz, bphi in points: . 422) nil (465 . 473) nil (q . -465) nil (# . -51) nil (nil rear-nonsticky nil 77 . 78) (nil fontified nil 52 . 78) (52 . 78) nil (50 . 52) nil (50 . 52) nil (nil rear-nonsticky nil 49 . 50) (nil fontified nil 11 . 50) (11 . 50) nil (58 . 61) nil (m . -58) (p . -59) (l . -60) nil (47 . 54) nil (590 . 593) nil (f . -590) (i . -591) (g . -592) nil (584 . 585) nil (614 . 615) nil (nil rear-nonsticky nil 680 . 681) (nil fontified nil 614 . 681) (614 . 681) nil (613 . 614) nil (nil rear-nonsticky nil 612 . 613) (nil fontified nil 584 . 613) (584 . 613) nil (390 . 391) nil (385 . 390) nil (378 . 385) nil (366 . 378) nil (, . -366) nil (360 . 367) nil (355 . 360) nil (  . -355) nil (347 . 356) nil (345 . 347) nil (301 . 310) nil ([ . -335) (0 . -336) (] . -337) nil (338 . 339) nil () . -338) nil (326 . 330) nil (311 . 326) nil (( . -311) nil (301 . 302) nil (, . -299) nil (300 . 301) nil (177 . 184) nil (i . -297) nil (294 . 298) nil (     . -294) nil (293 . 298) nil (apply activate-cursor-for-undo 14) (212 . 213) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) (231 . 232) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) (250 . 251) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) (269 . 270) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) (288 . 289) (apply deactivate-cursor-after-undo 18) (193 . 194) nil (apply activate-cursor-for-undo 14) (  . 211) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) (  . 231) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) (  . 251) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) (  . 271) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) (l . 291) (apply deactivate-cursor-after-undo 18) (  . 193) (apply activate-cursor-for-undo 14) (  . 212) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) (  . 233) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) (  . 254) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) (  . 275) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) (p . 296) (apply deactivate-cursor-after-undo 18) (  . 193) (apply activate-cursor-for-undo 14) (  . 213) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) (  . 235) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) (  . 257) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) (  . 279) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) (s . 301) (apply deactivate-cursor-after-undo 18) (  . 193) (apply activate-cursor-for-undo 14) (
 . 214) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) (
 . 237) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) (
 . 260) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) (
 . 283) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) (
 . 306) (apply deactivate-cursor-after-undo 18) (
 . 193) (apply activate-cursor-for-undo 14) () . 215) (apply deactivate-cursor-after-undo 14) (apply activate-cursor-for-undo 15) () . 239) (apply deactivate-cursor-after-undo 15) (apply activate-cursor-for-undo 16) () . 263) (apply deactivate-cursor-after-undo 16) (apply activate-cursor-for-undo 17) () . 287) (apply deactivate-cursor-after-undo 17) (apply activate-cursor-for-undo 18) () . 311) (apply deactivate-cursor-after-undo 18) () . 193) nil (193 . 194) nil () . -193) nil (apply activate-cursor-for-undo 8) (200 . 201) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (222 . 223) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (245 . 246) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (268 . 269) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (291 . 292) (apply deactivate-cursor-after-undo 12) (178 . 179) (apply activate-cursor-for-undo 8) (198 . 199) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (219 . 220) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (241 . 242) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (263 . 264) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (285 . 286) (apply deactivate-cursor-after-undo 12) (177 . 178) nil (apply activate-cursor-for-undo 8) (l . 197) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (l . 219) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (l . 242) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (l . 265) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (l . 288) (apply deactivate-cursor-after-undo 12) (l . 177) (apply activate-cursor-for-undo 8) (f . 198) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (f . 221) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (f . 245) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (f . 269) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (f . 293) (apply deactivate-cursor-after-undo 12) (f . 177) (apply activate-cursor-for-undo 8) (( . 199) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (( . 223) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (( . 248) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (( . 273) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (( . 298) (apply deactivate-cursor-after-undo 12) (( . 177) nil (apply activate-cursor-for-undo 8) (print . 200) (apply deactivate-cursor-after-undo 8) (apply activate-cursor-for-undo 9) (print . 229) (apply deactivate-cursor-after-undo 9) (apply activate-cursor-for-undo 10) (print . 259) (apply deactivate-cursor-after-undo 10) (apply activate-cursor-for-undo 11) (print . 289) (apply deactivate-cursor-after-undo 11) (apply activate-cursor-for-undo 12) (print . 319) (apply deactivate-cursor-after-undo 12) (print . 177) nil (
 . 173) nil (    print(line) . 173) nil (
 . 366) nil (    print(line[38:46]) . 366) nil (361 . 362) nil (5 . -361) nil (360 . 362) nil (383 . 384) nil (5 . -383) nil (383 . 384) nil (6 . 383) nil (
     . 384) nil (5 . 389) (385 . 389) (                . 385) nil (385 . 400) (     . 385) (389 . 390) nil (384 . 389) nil (383 . 384) nil (7 . -383) nil (383 . 384) nil (6 . -383) nil (383 . 384) nil (5 . -383) nil (383 . 384) nil (4 . -383) nil (383 . 384) nil (5 . -383) nil (383 . 384) nil (1 . -383) nil (379 . 384) nil (2 . -379) (9 . -380) (: . -381) (3 . -382) (8 . -383) nil (357 . 360) nil (0 . -357) (: . -358) (6 . -359) nil (327 . 332) nil (0 . -327) (: . -328) (6 . -329) nil (381 . 382) nil (7 . -381) nil (380 . 382) nil (2 . -380) (9 . -381) nil (378 . 379) nil (0 . -378) nil (300 . 302) nil (6 . -300) nil (297 . 299) nil (0 . -297) nil (379 . 380) nil (8 . -379) nil (379 . 380) nil (7 . -379) nil (379 . 380) nil (0 . -379) nil (375 . 377) nil (1 . -375) (3 . -376) nil (# . 246) nil (271 . 273) nil (1 . -271) (5 . -272) nil (379 . 381) nil (1 . -379) (9 . -380) nil (379 . 381) nil (2 . -379) (0 . -380) nil (380 . 381) nil (1 . -380) nil (380 . 381) nil (2 . -380) nil (380 . 381) nil (3 . -380) nil (380 . 381) nil (5 . -380) nil (380 . 381) nil (0 . -380) nil (379 . 381) nil (1 . -379) (7 . -380) nil (246 . 247) nil (379 . 380) nil (8 . -379) nil (377 . 380) nil (: . 375) (6 . 375) nil (270 . 272) nil (6 . -270) (1 . -271) nil (271 . 272) nil (268 . 269) nil (4 . -268) nil (267 . 269) nil (0 . -267) nil (240 . 242) nil (6 . -240) nil (238 . 239) nil (0 . -238) nil (375 . 376) nil (2 . -375) nil (375 . 376) nil (0 . -375) nil (374 . 376) nil (6 . -374) nil (372 . 373) nil (0 . -372) nil (nil fontified nil 356 . 357) (nil fontified nil 344 . 356) (nil fontified nil 339 . 344) (nil fontified nil 338 . 339) (nil fontified nil 333 . 338) (nil fontified nil 329 . 333) (329 . 357) nil (nil fontified nil 328 . 329) (nil fontified nil 316 . 328) (nil fontified nil 311 . 316) (nil fontified nil 310 . 311) (nil fontified nil 305 . 310) (nil fontified nil 301 . 305) (301 . 329) nil (nil fontified nil 300 . 301) (nil fontified nil 288 . 300) (nil fontified nil 283 . 288) (nil fontified nil 282 . 283) (nil fontified nil 277 . 282) (nil fontified nil 273 . 277) (273 . 301) nil (nil fontified nil 272 . 273) (nil fontified nil 260 . 272) (nil fontified nil 255 . 260) (nil fontified nil 254 . 255) (nil fontified nil 249 . 254) (nil fontified nil 245 . 249) (245 . 273) nil (nil fontified nil 244 . 245) (nil fontified nil 232 . 244) (nil fontified nil 227 . 232) (nil fontified nil 226 . 227) (nil fontified nil 221 . 226) (nil fontified nil 217 . 221) (217 . 245) nil (nil fontified nil 216 . 217) (nil fontified nil 204 . 216) (nil fontified nil 199 . 204) (nil fontified nil 198 . 199) (nil fontified nil 193 . 198) (nil fontified nil 189 . 193) (189 . 217) nil (
 . 189) nil (    print(float(line[0:6])) . 189) nil () . -237) nil (float( . -227) nil (nil rear-nonsticky nil 244 . 245) (nil fontified nil 244 . 245) (nil fontified nil 232 . 244) (nil fontified nil 227 . 232) (nil fontified nil 226 . 227) (nil fontified nil 221 . 226) (nil fontified nil 217 . 221) (217 . 245) nil (nil rear-nonsticky nil 216 . 217) (nil fontified nil 216 . 217) (nil fontified nil 204 . 216) (nil fontified nil 199 . 204) (nil fontified nil 198 . 199) (nil fontified nil 193 . 198) (nil fontified nil 189 . 193) (189 . 217) nil (
 . 189) nil (    print(float(line[0:6])) . 189) nil (214 . 215) nil (199 . 205) nil (206 . 207) nil (5 . -206) nil (228 . 229) nil (210 . 220) nil (205 . 209) nil (, . -205) nil (193 . 206) nil (188 . 193) nil (182 . 188) nil (  . -182) nil (177 . 183) nil (p . -177) (a . -178) (s . -179) (s . -180) nil (166 . 170) nil (165 . 167) nil (171 . 175) nil (166 . 171) nil (165 . 166) nil (171 . 174) nil (156 . 165) nil (p . -156) nil (152 . 157) nil (     . -152) nil (152 . 156) (     . 151) (150 . 156) nil (. . -156) (m . -157) (a . -158) nil (156 . 159) nil (147 . 148) nil (5 . -147) nil (147 . 150) nil (146 . 147) nil (147 . 152) nil (     . -147) nil (146 . 151) nil (
 . -146) nil (     . -147) nil (146 . 151) nil (145 . 146) nil (124 . 145) nil (119 . 124) nil (112 . 119) nil (65 . 69) nil (l . -65) (i . -66) (n . -67) (e . -68) nil (65 . 69) nil (w . -65) (i . -66) (t . -67) (h . -68) nil (110 . 112) nil (nil rear-nonsticky nil 109 . 110) (nil fontified nil 83 . 110) (83 . 110) nil (82 . 83) nil (nil rear-nonsticky nil 81 . 82) (nil fontified nil 76 . 82) (nil :pm-span (body 63 75 #s(pm-inner-overlay-chunkmode #s(pm-inner-overlay-chunkmode #s(pm-inner-overlay-chunkmode unbound ein-input-cell nil nil poly-fallback-mode t 2 0 0 t t t 2 (poly-ein-undo-damage poly-ein-init-input-cell) nil nil nil 0 nil nil poly-head-tail-mode poly-head-tail-mode unbound unbound bold nil nil nil unbound) ein-input-cell: unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound) ein-input-cell:: unbound python-mode unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound  *ein: http://127.0.0.1:8888/Untitled.ipynb*[python] unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound unbound)) 76 . 82) (nil :pm-mode python-mode 76 . 82) (nil fontified t 76 . 82) (nil face font-lock-string-face 76 . 82) (76 . 82) nil (65 . 76) nil (36 . 54) nil (30 . 36) nil (29 . 30) nil (22 . 29) nil (t . -22) nil (11 . 23) (t . 0)) (fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e d268e d268e d268e d268e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e d268e 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 3ec17 fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e fa03e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3473e 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0 3d6d0) | 2
```
This happened when I tried to execute a cell. I'm using matplotlib in  a notebook running on ein from 3f30422f0a7f5d06272169bfd12d61274fec6651.

**Comment by dickmao** [comment]
May I see the definitions of `activate-cursor-for-undo` and `deactivate-cursor-after-undo` and the hooks (presumably in your .emacs) that reference them?

**Comment by dickmao** [comment]
I preferred fixing this by unilaterally turning off multiple cursors, but that seems more difficult than just inserting `(multiple-cursors-mode -1)`.

So I implemented a fudge in 506fcb5.  I tested it very lightly.

One certain way to avoid this bug is to eschew multiple cursors in EIN.

Your feedback is consistently valuable!

**Comment by veprbl** [comment]
@dickmao Thank you for your work on this. I must admit I am slightly confused about this issue. I was using multiple-cursors independently from ein and, without knowing how it really works, I naively assumed that the future is implemented using only simple crude tricks that can not affect operation of the major modes such as ein. I guess, I was wrong. I'm not entirely sure what are the implications of 506fcb5f985a77b77f29f4978446f1b218a2fd87 yet, I might need to test it myself and get back to you.
